### PR TITLE
add fog piece, remove some chess checks

### PR DESCRIPTION
**Pull request by JiriH94**
Hey,  some possible inspiration, it's not fully done and had to comment out the tests etc. if I'll find time I can continue on some changes, but I am not sure what direction you want to take :)

### CODE REVIEW

File: Chess-Challenge/resources/Pieces.png
Comment (ak47andrew) [inline Chess-Challenge/resources/Pieces.png:1]: 1. Change question mark with something more suitable (chess.com style for example)
2. Also it's shifted for whatever reason
Screenshot_20241104_214718

File: Chess-Challenge/src/API/Board.cs
Comment (ak47andrew) [inline Chess-Challenge/src/API/Board.cs:1]: Remove ForceSkipTurn bc it's no longer needed after SkipTurn

File: Chess-Challenge/src/Framework/Application/Helpers/Tester.cs
Comment (ak47andrew) [inline Chess-Challenge/src/Framework/Application/Helpers/Tester.cs:1]: It's just deeply funny to me that you commented out this file completely. Not required, but better to add some

File: Chess-Challenge/src/Framework/Chess/Move Generation/MoveGenerator.cs
Comment (ak47andrew) [inline Chess-Challenge/src/Framework/Chess/Move Generation/MoveGenerator.cs:1]: I don't know why, but checks and stalemate is still a thing. Here's some test fens that should be run with MyBot and will raise IndexError:
`3k4/3r3p/p1p4b/3pp1p1/n7/8/2b5/K1n5 w - - 4 79`
`8/8/3nb3/1p1p1p2/p7/1rpk4/3r4/2K4n w - - 6 151`
`5k2/2b2n1r/8/5p2/1p2p3/p7/1r6/K7 w - - 4 101`

**Review by ak47andrew — CHANGES_REQUESTED** [review]

File: Chess-Challenge/src/API/Board.cs
Comment (JiriH94) [reply Chess-Challenge/src/API/Board.cs:1]: deleted, also made the SkipTurn return void

File: Chess-Challenge/src/Framework/Chess/Move Generation/MoveGenerator.cs
Comment (ak47andrew) [reply Chess-Challenge/src/Framework/Chess/Move Generation/MoveGenerator.cs:1]: Waiting for this one to continue working on a project. Probably will submit pull request on this issue or so, but not really sure

Comment (JiriH94) [reply Chess-Challenge/src/Framework/Chess/Move Generation/MoveGenerator.cs:1]: should be fixed now

**Review by ak47andrew — APPROVED** [review]
Sounds good to me! Haven't checked everything yet, but a quick inspection is done.
Merge, thanks for a big chunk of work done!

**Pull request merged**